modules/root/root_api.rb

```diff
@@ -25,4 +27,19 @@ class Proxy::RootApi < Sinatra::Base
       log_halt 400, e
     end
   end
+
+  get "/events" do
+    begin
+      content_type :json
+
+      events = []
+      from_serial = params[:from_serial].to_i rescue 0
+      ::Proxy::Events::Buffer.instance.iterate_ascending do |event|
+        events << event.to_h if event && from_serial == 0 || (event && from_serial > 0 && event.serial >= from_serial)
```

Comment:
nitpick: maybe extract this logic outside here?
regardless, if from_serial is zero, you probably don't need to compare it for every event?